src/features/metadata-instance-editor/fields/DateField.js

```diff
@@ -18,29 +19,41 @@ type Props = {
     onRemove: (key: string) => void,
 };
 
-const DateField = ({ dataKey, dataValue, displayName, description, intl, onChange, onRemove }: Props) => (
-    <DatePicker
-        className="metadata-instance-editor-field-date"
-        dateFormat="utcISOString"
-        description={description}
-        displayFormat={{
-            day: 'numeric',
-            month: 'long',
-            year: 'numeric',
-        }}
-        hideOptionalLabel
-        label={displayName}
-        onChange={(date: Date, isoDate: string) => {
-            if (isoDate) {
-                onChange(dataKey, isoDate);
-            } else {
-                onRemove(dataKey);
-            }
-        }}
-        placeholder={intl.formatMessage(messages.metadataFieldSetDate)}
-        value={typeof dataValue === 'string' ? convertISOStringToUTCDate(dataValue) : undefined}
-    />
-);
+const DateField = ({ dataKey, dataValue, displayName, description, intl, onChange, onRemove }: Props) => {
+    // if the JS engine thinks there's an invalid date, convert to known format if possible
+    let parsedDataValue = dataValue;
+
+    try {
+        parsedDataValue =
+            !isNaN(dataValue) && typeof dataValue === 'string' ? convertISOStringtoRFC3339String(dataValue) : dataValue;
+    } catch (err) {
+        // no-op; keep the existing dataValue
```

Comment:
Can we just assign the default value here rather than having the comment? `parsedDataValue = dataValue`?

Reply:
I could do that

Reply:
Actually, there are some factors that create the following scenario:

- I can't define a `let` and `var` inside the try-catch due to variable scope and "no-var"/"vars-on-top" respectively
- if I put a `let` at the top, I could define it with no default, and set it only in the try-catch. this adds one extra line, and achieves the same behavior as what I have already
- I don't think it's beneficial to both define `let parseDataValue = dataValue` at line 24, then set it again in the `catch` to the same value

I'm inclined to leave this one as-is

Reply:
removed this, and make the checks simpler